wgpu-types/src/lib.rs

```diff
@@ -128,6 +123,39 @@ impl From<Backend> for Backends {
     }
 }
 
+impl Backends {
+    /// Generates a set of backends from a comma separated list of case-insensitive backend names.
+    ///
+    /// Whitespace is stripped, so both 'gl, dx12' and 'gl,dx12' are valid.
+    //
+    /// Names:
+    /// - vulkan = "vulkan" or "vk"
+    /// - dx12   = "dx12" or "d3d12"
+    /// - dx11   = "dx11" or "d3d11"
+    /// - metal  = "metal" or "mtl"
+    /// - gles   = "opengl" or "gles" or "gl"
+    /// - webgpu = "webgpu"
+    pub fn from_comma_list(string: &str) -> Option<Self> {
+        let mut backends = Self::empty();
+        for backend in string.to_lowercase().split(',') {
+            backends |= match backend.trim() {
+                "vulkan" | "vk" => Self::VULKAN,
+                "dx12" | "d3d12" => Self::DX12,
+                "dx11" | "d3d11" => Self::DX11,
+                "metal" | "mtl" => Self::METAL,
+                "opengl" | "gles" | "gl" => Self::GL,
+                "webgpu" => Self::BROWSER_WEBGPU,
+                _ => continue,
+            }
+        }
+        if backends == Self::empty() {
+            None
+        } else {
+            Some(backends)
```

Comment:
actually, why are we doing this? If the user has a string, we should just assume they know what they are doing. Let's return the bitflags instead of an option

Reply:
I changed this in favor of logging when it's empty, as it's _probably_ not intended.